src/kOS/Module/kOSProcessor.cs

```diff
@@ -528,8 +528,15 @@ public void SetMode(ProcessorModes newProcessorMode)
                 switch (newProcessorMode)
                 {
                     case ProcessorModes.READY:
-                        if ((ProcessorMode == ProcessorModes.STARVED || ProcessorMode == ProcessorModes.OFF) && shared.Cpu != null)
-                            shared.Cpu.Boot();
+                        if (Config.Instance.StartOnArchive)
```

Comment:
You're right . I missed the line above saying :
```if (ProcessorMode != newProcessorMode)```
It was in the bit of the github diff output that was folded away and hidden.